### ISSUE TITLE
custom-error-definition does not understand class properties

### ISSUE THREAD

**Issue by futpib** [sindresorhus/eslint-plugin-unicorn#130]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2460%20Funded-%2300A156.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/130)

The following code is reported, even though it does set the `name` property to `ValidationError`.

```js
export class ValidationError extends Error {
	name = 'ValidationError';
}
```

```
error  The `name` property should be set to `ValidationError`  unicorn/custom-error-definition
```

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $60.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://issuehunt.io/u/issuehunt) ($60.00)

### Submitted pull Requests
- [#673 `custom-error-definition`: Support class properties](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/pull/673)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

**Comment by sindresorhus** [comment]
Are you using `babel-eslint`? I don't think ESLint support class properties yet.

**Comment by futpib** [comment]
Yes, I'm using `babel-eslint`.

**Comment by sindresorhus** [comment]
I don't think we're interested in adding a separate code path for when `babel-eslint` is used. We can't support it by default as ESLint doesn't expose the new AST nodes needed.

When https://github.com/tc39/proposal-class-fields reaches stage 4, ESLint will support it, and we can too. I'll keep this issue open until then.

**Comment by tinovyatkin** [comment]
Node 12 supports class properties out of the box now.

**Comment by felixfbecker** [comment]
I have the same problem with `typescript-eslint`

**Comment by sindresorhus** [comment]
Alright, let's try to support this for `babel-eslint` and `typescript-eslint`.

**Comment by tinovyatkin** [comment]
ESLint 6.x while targeting ES2020 supports it's natively too.

**Comment by karuppiah7890** [comment]
Can I take a shot at this? I'm new to this whole repo 😅 . But I'm willing to try, and also get paid for the first time for an OSS contribution 😁

**Comment by sindresorhus** [comment]
@karuppiah7890 This is not an easy issue and it requires some JS and AST experience.

Check out https://issuehunt.io and https://github.com/search?o=desc&q=user%3Asindresorhus+user%3Axojs+user%3Aavajs+user%3Achalk++is%3Aissue+is%3Aopen+commenter%3AIssuehuntBot&s=created&type=Issues for other issues ;)

**Comment by maximelkin** [comment]
Can I take this?
